### PR TITLE
fix(HeaderMenu) : render menuItem as anchor or button

### PR DESCRIPTION
**Pull request by cheikhgwane**
Fix https://github.com/BLSQ/openhexa-team/issues/92 

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- add linkTo props to MenuItem component

## How/what to test

Click on the user menu dropdown and check if the items can be opened in a new tab.

## Screenshots / screencast
image

### CODE REVIEW

File: src/core/components/Menu.tsx [hunk 1]
```diff
@@ -63,21 +63,29 @@ const Item = ({
   activeClassName = MenuClasses.ActiveItem,
   onClick,
   className = MenuClasses.Item,
+  linkTo,
 }: {
   children: ReactNode;
   onClick?: ((event: { preventDefault: Function }) => void) | undefined;
   className?: string;
   activeClassName?: string;
+  linkTo?: string;
```
Comment (qgerome) [inline src/core/components/Menu.tsx:72]: I'd prefer to use the same than link: `href` WDYT ?

File: src/core/components/Menu.tsx [hunk 2]
```diff
@@ -63,21 +63,29 @@ const Item = ({
...
+    {({ active }) =>
+      onClick ? (
+        <button
+          onClick={onClick}
+          className={clsx(className, active && activeClassName)}
+        >
+          {children}
+        </button>
+      ) : (
+        <a href={linkTo} className={clsx(className, active && activeClassName)}>
```
Comment (qgerome) [inline src/core/components/Menu.tsx:84]: You should use `Link` instead of a raw anchor

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/core/components/Menu.tsx (hunk 2 above)
Comment (cheikhgwane) [reply src/core/components/Menu.tsx:84]: Got it .

File: src/core/components/Menu.tsx (hunk 1 above)
Comment (cheikhgwane) [reply src/core/components/Menu.tsx:72]: Yes, sounds good.

File: src/core/components/Menu.tsx [hunk 3]
```diff
@@ -63,21 +64,32 @@ const Item = ({
...
+      onClick ? (
+        <button
+          onClick={onClick}
+          className={clsx(className, active && activeClassName)}
+        >
+          {children}
+        </button>
+      ) : (
+        <Link
+          href={href || ""}
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/core/components/Menu.tsx:86]: Okay

File: src/core/components/Menu.tsx [hunk 4]
```diff
@@ -58,25 +59,46 @@ const Menu = ({ label, trigger, className, children }: MenuProps) => {
   );
 };
 
+type ItemProps = {
+  children: ReactNode;
+  className?: string;
+  activeClassName?: string;
+} & (
+  | {
+      onClick: ((event: { preventDefault: Function }) => void) | undefined;
```
Comment (qgerome) [inline src/core/components/Menu.tsx:68]: You can remove the `| undefined`

**Review by qgerome — CHANGES_REQUESTED** [review]

**Pull request merged**